### PR TITLE
docs: clarify workdir and cmd attributes

### LINKED ISSUE #921: [Bug]: Running Nextjs example in Docker fails with `/app/apps/alpha/bin.runfiles/__main__/apps/alpha/bin.sh: line 379: cd: apps/alpha/dist: No such file or directory`
### What happened?

I have been working on setting up the [nextjs example](https://github.com/aspect-build/bazel-examples/tree/main/next.js) in this repository to work on docker. 

I have combined the image example [js_image](https://github.com/aspect-build/rules_js/tree/main/e2e/js_image_oci) with the [nextjs docker example implementation](https://github.com/vercel/next.js/tree/canary/examples/with-docker) to successfully create a docker image.

I get the error below when I try to run the docker image:

```
/app/apps/alpha/bin.runfiles/__main__/apps/alpha/bin.sh: line 379: cd: apps/alpha/dist: No such file or directory
```

I have a fork of the repository with the changes required to build the image [here](https://github.com/noelenwenede/bazel-examples/tree/main/next.js).

Side Note:
Inspecting the directories in the generated image `apps/alpha/dist` directory exist inside  the`/app/apps/alpha/bin.runfiles/__main__` (I believe the issue may be related to `BAZEL_BINDIR`).  

Running the docker image in https://github.com/aspect-build/rules_js/tree/main/e2e/js_image_oci here works fine. So I'm unsure where the error comes from.

### Version

Development (host) and target OS/architectures: MacBook Pro Monterey (M1)

Output of `bazel --version`: bazel 6.0.0-homebrew

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file:
1. rules_js-1.20.1
2. rules_ts-1.3.1
3. rules_python-0.14.0
4. rules_pkg-0.8.0
5. rules_oci-0.3.1

Language(s) and/or frameworks involved:
1. next.js
2. typescript

### How to reproduce

```shell
1. Clone https://github.com/noelenwenede/bazel-examples/tree/main/next.js
2.  cd next.js
3. bazel build //apps/alpha:next_tarball
4. docker load --input $(bazel cquery --output=files //apps/alpha:next_tarball)
5. docker run -p 3000:3000 -t -i nextjs/js:latest
```

### Any other information?

_No response_

### Fund our work

- [ ] Sponsor our open source work by donating a [bug bounty](https://opencollective.com/aspect-build/)

### PR DESCRIPTION
**Pull request by thesayyn**
# Description

Please include a summary of the changes and the related issue (if any). Please also include relevant motivation and context.

Fixes #921

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature or functionality (change which adds functionality)
- [ ] Style (white-space, formatting, etc...)
- [ ] Refactor (a code change that neither fixes a bug or adds a new feature)
- [ ] Performance (a code change that improves performance)
- [x] Documentation (updates to documentation or READMEs)
- [ ] Chore (any other change that doesn't affect source or test files)

# Properties of change

- [ ] Breaking change (this change will cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Test plan

How has this been tested?

- [ ] Covered by existing tests cases
- [ ] New test cases added
- [ ] Manual testing

If part of the test plan included manual testing, please provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

### CODE REVIEW

File: js/private/js_image_layer.bzl [hunk 1]
```diff
@@ -18,9 +18,19 @@ load("@bazel_skylib//lib:paths.bzl", "paths")
 
 _DOC = """Create container image layers from js_binary targets.
 
-js_image_layer supports transitioning to specific platform for cross-compiling.
+By design, js_image_layer doesn't have any preference over rule which assembles the container image.
```
Comment (jbedard) [inline js/private/js_image_layer.bzl:21]:
```suggestion
By design, js_image_layer doesn't have any preference over which rule assembles the container image. 
```

File: js/private/js_image_layer.bzl [hunk 2]
```diff
@@ -18,9 +18,19 @@ load("@bazel_skylib//lib:paths.bzl", "paths")
 
 _DOC = """Create container image layers from js_binary targets.
 
-js_image_layer supports transitioning to specific platform for cross-compiling.
+By design, js_image_layer doesn't have any preference over rule which assembles the container image. 
+However, this comes with a few quirks; the downstream rule, `oci_image` or `container_image` in this case,
+has to set proper `workdir` and `cmd` to make the container work.
```
Comment (jbedard) [inline js/private/js_image_layer.bzl:23]: Should we clarify what a "proper" workdir is? I'm not sure what "proper" means in this case...

File: js/private/js_image_layer.bzl [hunk 3]
```diff
@@ -18,9 +18,19 @@ load("@bazel_skylib//lib:paths.bzl", "paths")
...
 
-A partial example using rules_oci with transition to linux/amd64 platform.
+Usually, `cmd` needs to be set to **/** `[ root of <js_image_layer> ]` **/** `[ relative path to BUILD file from WORKSPACE or package_name() ] / [ name of <js_binary> ]`,
```
Comment (jbedard) [inline js/private/js_image_layer.bzl:25]:
```suggestion
Usually `cmd` needs to be set to **/** `[ root of <js_image_layer> ]` **/** `[ relative path to BUILD file from WORKSPACE or package_name() ] / [ name of <js_binary> ]`, 
```

I'm not sure what the bold "/"s mean.. is that a syntax or trying to highlight something?

File: js/private/js_image_layer.bzl [hunk 4]
```diff
@@ -18,9 +18,19 @@ load("@bazel_skylib//lib:paths.bzl", "paths")
...
 _DOC = """Create container image layers from js_binary targets.
 
-js_image_layer supports transitioning to specific platform for cross-compiling.
+By design, js_image_layer doesn't have any preference over rule which assembles the container image. 
+However, this comes with a few quirks; the downstream rule, `oci_image` or `container_image` in this case,
+has to set proper `workdir` and `cmd` to make the container work.
 
-A partial example using rules_oci with transition to linux/amd64 platform.
+Usually, `cmd` needs to be set to **/** `[ root of <js_image_layer> ]` **/** `[ relative path to BUILD file from WORKSPACE or package_name() ] / [ name of <js_binary> ]`, 
+unless you have some other launcher script that invokes entrypoint of the `js_binary`.
```
Comment (jbedard) [inline js/private/js_image_layer.bzl:26]:
```suggestion
unless you have a launcher script that invokes the entry_point of the `js_binary`.
```

File: js/private/js_image_layer.bzl [hunk 5]
```diff
@@ -18,9 +18,19 @@ load("@bazel_skylib//lib:paths.bzl", "paths")
 
 _DOC = """Create container image layers from js_binary targets.
 
-js_image_layer supports transitioning to specific platform for cross-compiling.
+By design, js_image_layer doesn't have any preference over rule which assembles the container image. 
+However, this comes with a few quirks; the downstream rule, `oci_image` or `container_image` in this case,
```
Comment (jbedard) [inline js/private/js_image_layer.bzl:22]:
```suggestion
This means the downstream rule (`oci_image`, or `container_image` in this case) must set a proper `workdir` and `cmd` to for the container work.
```
? 🤷 

What do you mean by "in this case" though? The example below? Or `js_image_layer`? You could probably say "in this example" or "with js_image_layer"?

Comment (thesayyn) [reply js/private/js_image_layer.bzl:22]: the downstream rule, it should be `in most cases`

File: js/private/js_image_layer.bzl (hunk 2 above)
Comment (thesayyn) [reply js/private/js_image_layer.bzl:23]: that's what is done down below. starts with "Usually"

File: js/private/js_image_layer.bzl (hunk 3 above)
Comment (thesayyn) [reply js/private/js_image_layer.bzl:25]: those are not supposed to look this bad. I'll revert those.

File: js/private/js_image_layer.bzl [hunk 6]
```diff
@@ -18,9 +18,16 @@ load("@bazel_skylib//lib:paths.bzl", "paths")
 
 _DOC = """Create container image layers from js_binary targets.
 
-js_image_layer supports transitioning to specific platform for cross-compiling.
+By design, js_image_layer doesn't have any preference over which rule assembles the container image. 
+This means the downstream rule (`oci_image`, or `container_image` in this case) must set a proper `workdir` and `cmd` to for the container work.
+A proper `cmd` usually looks like /`[ root of <js_image_layer> ]`/`[ relative path to BUILD file from WORKSPACE or package_name() ]/[ name of <js_binary> ]`,
```
Comment (jbedard) [inline js/private/js_image_layer.bzl:23]: The `<` seems to be showing up as `&lt;` in the .md which doesn't seem correct? At least looks off in the github ui...

**Review by jbedard — APPROVED** [review]

File: js/private/js_image_layer.bzl (hunk 6 above)
Comment (thesayyn) [reply js/private/js_image_layer.bzl]: removed those. IDK why they don't work

File: js/private/js_image_layer.bzl [hunk 7]
```diff
@@ -18,9 +18,16 @@ load("@bazel_skylib//lib:paths.bzl", "paths")
 
 _DOC = """Create container image layers from js_binary targets.
 
-js_image_layer supports transitioning to specific platform for cross-compiling.
+By design, js_image_layer doesn't have any preference over which rule assembles the container image. 
+This means the downstream rule (`oci_image`, or `container_image` in this case) must set a proper `workdir` and `cmd` to for the container work.
+A proper `cmd` usually looks like /`[ root of js_image_layer ]`/`[ relative path to BUILD file from WORKSPACE or package_name() ]/[ name of js_binary ]`, 
+unless you have a launcher script that invokes the entry_point of the `js_binary` in a different path.
+On the other hand, `workdir` has to be set to `runfiles tree root` which would be exactly `cmd` **but with `.runfiles` suffix**. If `workdir` is not set correctly, some
+attributes such as `chdir` might not work properly.
```
Comment (LavaToaster) [inline js/private/js_image_layer.bzl:26]: The workspace name needs to be included in this path (either what's in WORKSPACE[.bazel] or `__main__` if unset).

Comment (LavaToaster) [reply js/private/js_image_layer.bzl:26]: Note: getting access to the workspace name in a macro is impossible.

**Pull request merged**